### DISCUSSION TITLE
How to handle MessageDialog buttons?

### DISCUSSION THREAD

**Discussion by hstig** [faaxm/spix#71] [General]
If a modal MessageDialog pops up, how can I address the buttons of this dialog.

**Comment by faaxm** [comment]
For now, Spix can only interact with elements in QML. If a system window/dialog is displayed, it cannot access any of its UI elements as they are not part of QML.

If you use Spix together with python scripting and use pyAutoGUI to control the mouse, you could try to use its locate function to find the buttons in the dialog:
https://pyautogui.readthedocs.io/en/latest/screenshot.html#the-locate-functions

**Comment by hstig** [comment]
Sorry for not being clear enough. By MessageDialog I mean a QML element: https://doc.qt.io/qt-6/qml-qtquick-dialogs-messagedialog.html.
